src/App.tsx

```diff
@@ -1,26 +1,286 @@
+    } finally {
+      setLoadingTodos(prev => prev.filter(loadingId => loadingId !== id));
+      inputRef.current?.focus();
+    }
+  };
+
+  // const handleToggleAllTodos = async () => {
+  //   const allCompleted = todos.every(todo => todo.completed);
+
+  //   try {
+  //     const updatedTodos = await Promise.all(
+  //       todos.map(async todo => {
+  //         const updatedTodo = { ...todo, completed: !allCompleted };
+
+  //         await editTodo(updatedTodo);
+
+  //         return updatedTodo;
+  //       }),
+  //     );
+
+  //     setTodos(updatedTodos);
+  //   } catch (error) {
+  //     showError('Unable to update todos');
+  //   }
+  // };
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      try {
+        const todosData = await getTodos();
+
+        setTodos(todosData);
+        updateClearCompletedState(todosData);
+      } catch (error) {
+        showError('Unable to load todos');
+      } finally {
+        setIsLoadingTodos(false);
+      }
+    };
+
+    fetchTodos();
+  }, [showError, updateClearCompletedState]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          <button
+            type="button"
+            className="todoapp__toggle-all"
+            data-cy="ToggleAllButton"
+            // onClick={handleToggleAllTodos}
+          />
+          <form onSubmit={handleFormSubmit}>
+            <input
+              data-cy="NewTodoField"
+              ref={inputRef}
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              value={input}
+              onChange={e => setInput(e.target.value)}
+              disabled={isInputDisabled}
+            />
+          </form>
+        </header>
+
+        {tempTodo ? (
+          <div className="todoapp__todo-item">
+            <span className="todoapp__todo-title">{tempTodo.title}</span>
+            <span className="loader"></span>
+          </div>
+        ) : (
+          ''
+        )}
+
+        {!isLoadingTodos && todos.length ? (
+          <TodoList
+            clearDisabled={isClearCompletedDesabled}
+            tempTodo={tempTodo}
+            deleteDisabled={loadingTodos}
+            deleteTodo={handleDeleteTodo}
+            todos={todos}
+            deleteCompleted={handleDeleteCompletedTodos}
+          />
+        ) : (
+          ''
+        )}
+      </div>
```

Comment:
better option(more readable)

```suggestion
        {tempTodo ?? (
          <div className="todoapp__todo-item">
            <span className="todoapp__todo-title">{tempTodo.title}</span>
            <span className="loader"></span>
          </div>
        )}

        {(!isLoadingTodos && todos.length) ?? (
          <TodoList
            clearDisabled={isClearCompletedDesabled}
            tempTodo={tempTodo}
            deleteDisabled={loadingTodos}
            deleteTodo={handleDeleteTodo}
            todos={todos}
            deleteCompleted={handleDeleteCompletedTodos}
          />
        )}
      </div>
```